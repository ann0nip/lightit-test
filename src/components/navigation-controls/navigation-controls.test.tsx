import { fireEvent, render, cleanup } from '@testing-library/react'
import NavigationControls from './navigation-controls.component';

const mockHandlerClick = jest.fn()

afterEach(cleanup);

describe('Button Component', () => {

    it('should take a snapshot', () => {
        const { asFragment } = render(<NavigationControls handleClick={mockHandlerClick} />)

        expect(asFragment()).toMatchSnapshot()
    })

    it('should run the onClick event for Zoom In button', () => {
        const { getByTestId } = render(<NavigationControls handleClick={mockHandlerClick} />)

        const buttonZoomIn = getByTestId('Zoom In')
        fireEvent.click(buttonZoomIn)
        expect(mockHandlerClick.mock.calls).toHaveLength(1)


    })
    it('should run the onClick event for Zoom Out button', () => {
        const { getByTestId } = render(<NavigationControls handleClick={mockHandlerClick} />)

        const buttonZoomOut = getByTestId('Zoom Out')
        fireEvent.click(buttonZoomOut)
        expect(mockHandlerClick.mock.calls).toHaveLength(1)


    })
    it('should run the onClick event for Flecha UP button', () => {
        const { getByTestId } = render(<NavigationControls handleClick={mockHandlerClick} />)

        const buttonFlechaUP = getByTestId('Flecha UP')
        fireEvent.click(buttonFlechaUP)
        expect(mockHandlerClick.mock.calls).toHaveLength(1)

    })
    it('should run the onClick event for Flecha Right button', () => {
        const { getByTestId } = render(<NavigationControls handleClick={mockHandlerClick} />)

        const buttonFlechaRight = getByTestId('Flecha Right')
        fireEvent.click(buttonFlechaRight)
        expect(mockHandlerClick.mock.calls).toHaveLength(1)

    })
    it('should run the onClick event for Flecha Left button', () => {
        const { getByTestId } = render(<NavigationControls handleClick={mockHandlerClick} />)

        const buttonFlechaLeft = getByTestId('Flecha Left')
        fireEvent.click(buttonFlechaLeft)
        expect(mockHandlerClick.mock.calls).toHaveLength(1)

    })
    it('should run the onClick event for Flecha Down button', () => {
        const { getByTestId } = render(<NavigationControls handleClick={mockHandlerClick} />)

        const buttonFlechaDown = getByTestId('Flecha Down')
        fireEvent.click(buttonFlechaDown)
        expect(mockHandlerClick.mock.calls).toHaveLength(1)

    })
})