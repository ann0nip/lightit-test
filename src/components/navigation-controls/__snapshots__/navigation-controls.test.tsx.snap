// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-row border-[#979797] cursor-pointer m-2"
    data-testid="navigation-controls"
  >
    <div
      class="bg-white rounded-md flex flex-col flex-wrap m-2"
    >
      <div
        class="flex flex-1 items-center border-b"
        data-name="Zoom In"
        data-testid="Zoom In"
      >
        <img
          alt="Zoom In"
          src="ZoomIn.png"
        />
      </div>
      <div
        class="flex flex-1 items-center"
        data-name="Zoom Out"
        data-testid="Zoom Out"
      >
        <img
          alt="Zoom Out"
          src="ZoomOut.png"
        />
      </div>
    </div>
    <div
      class="w-24 bg-white h-24 rounded-md flex flex-wrap m-2"
    >
      <div
        class="w-8 flex"
      />
      <div
        class="w-8 flex"
        data-name="Flecha UP"
        data-testid="Flecha UP"
      >
        <img
          alt="Flecha UP"
          src="Flecha.png"
        />
      </div>
      <div
        class="w-8 flex"
      />
      <div
        class="w-8 flex -rotate-90"
        data-name="Flecha Right"
        data-testid="Flecha Right"
      >
        <img
          alt="Flecha Right"
          src="Flecha.png"
        />
      </div>
      <div
        class="w-8 flex"
      />
      <div
        class="w-8 flex rotate-90"
        data-name="Flecha Left"
        data-testid="Flecha Left"
      >
        <img
          alt="Flecha Left"
          src="Flecha.png"
        />
      </div>
      <div
        class="w-8 flex"
      />
      <div
        class="w-8 flex rotate-180"
        data-name="Flecha Down"
        data-testid="Flecha Down"
      >
        <img
          alt="Flecha Down"
          src="Flecha.png"
        />
      </div>
      <div
        class="w-8 flex"
      />
    </div>
  </div>
</DocumentFragment>
`;
